src/pages/instances/actions/ReconnectTerminalBtn.tsx

```diff
@@ -10,23 +12,64 @@ interface Props {
 const ReconnectTerminalBtn: FC<Props> = ({ onFinish }) => {
   const [isModal, setModal] = useState(false);
 
-  const confirmModal = (result: LxdTerminalPayload) => {
-    onFinish(result);
+  const closeModal = () => {
     setModal(false);
   };
 
   const openModal = () => {
     setModal(true);
   };
 
+  const TerminalSchema = Yup.object().shape({
+    command: Yup.string().required("This field is required"),
+    environment: Yup.array().of(
+      Yup.object().shape({
+        key: Yup.string(),
+        value: Yup.string(),
+      })
+    ),
+    user: Yup.number(),
+    group: Yup.number(),
+  });
+
+  const formik = useFormik({
+    initialValues: {
+      command: "bash",
+      environment: [
+        {
+          key: "TERM",
+          value: "xterm-256color",
+        },
+        {
+          key: "HOME",
+          value: "/root",
+        },
+      ],
+      user: 0,
+      group: 0,
+    },
```

Comment:
I see these are defined in multiple places ([here](https://github.com/canonical/lxd-ui/blob/b25dc0b887787e3d491c53cf95f84b7a2e232d73/src/pages/instances/InstanceTerminal.tsx#L24)) and I was wondering, can we extract these shared props so that we can easily reuse them? For the `defaultPayload`, we can probably spread the additional properties that it has.
Also, giving it a type, or at least to the objects inside the `environment` array, would probably not be a bad idea.

Reply:
The formats are different. Here we have a format for the form, in the other case we have the request payload form. We have some logic 5 lines below to transform the form-type into the request-type.

Thinking about it -- We can probably move that logic into the API connector and pass an object with the form-type to it. Then we can also share the default object.